cfme/infrastructure/provider/__init__.py

```diff
@@ -286,10 +292,74 @@ def hosts(self):
         return result
 
 
+@navigator.register(Provider, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Compute', 'Infrastructure', 'Providers')(None)
+
+
+@navigator.register(Provider, 'Add')
+class Add(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Add a New Infrastructure Provider')
+
+
+@navigator.register(Provider, 'Discover')
+class Discover(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Discover Infrastructure Providers')
+
+
+@navigator.register(Provider, 'Details')
+class Details(CFMENavigateStep):
+    def prerequisite(self):
+        NavigateToSibling('All')
+        # Make sure no quadicons are selected and default to grid view
+        # Can we assume these two things at this level? (comment to be removed before merge)
+        # If this should be actively set, a utility function will be needed to shorten this
+        tb.select("Grid View")
```

Comment:
I think these should be added to the ```step``` method of the ```All``` destination. They are effectively resetters.